### PR TITLE
Tentative fix for compiler warning

### PR DESCRIPTION
**Pull request by bluescarni**

### CODE REVIEW

**Comment by bluescarni** [comment]
There's an unused variable in a function in ``xsimd_avx_int64.hpp``. I just removed the unused variable and renamed the other vars accordingly, but I have no idea if that's correct :)

There are also some ``static inline`` functions in ``xsimd_rem_pio2.hpp`` clang is complaining about, with messages such as:

```
/home/yardbird/.local/include/xsimd/math/xsimd_rem_pio2.hpp:449:29: warning: 'static' function '__ieee754_rem_pio2' declared in header file should be declared
      'static inline' [-Wunneeded-internal-declaration]
        static std::int32_t __ieee754_rem_pio2(double x, double* y)
```

I can apply the suggested change in the PR if it makes sense.

File: include/xsimd/types/xsimd_avx_int64.hpp [hunk 1]
```diff
@@ -648,10 +648,9 @@ namespace xsimd
 #else
         XSIMD_SPLIT_AVX(rhs);
         __m128i tmp1 = _mm_shuffle_epi32(rhs_low, 0x0E);
-        __m128i tmp2 = _mm_add_epi64(tmp1, rhs_low);
-        __m128i tmp3 = _mm_shuffle_epi32(rhs_high, 0x0E);
-        __m128i tmp4 = _mm_add_epi64(tmp3, rhs_high);
-        __m128i res = _mm_add_epi64(tmp3, tmp4);
```
Comment (JohanMabille) [inline include/xsimd/types/xsimd_avx_int64.hpp]: I think the problem is the last line, the fix should be

```cpp
__m128i res = _mm_add_epi64(tmp2, tmp4);
```

**Comment by JohanMabille** [comment]
Thanks for catching this! See my review for the fix of the unused variable, about the warnings regarding the static functions, I think we can drop the `static` specifier and use `inline` only.

**Comment by bluescarni** [comment]
No problem, thanks for the great library. I updated the PR.

**Pull request merged**